src/js/index.js

```diff
@@ -1,9 +1,33 @@
 import '../css/index.css';
 import { jokeFetcher } from './jokeFetcher';
 import { renderJoke } from './renderJoke';
+import { getNextJokeIdx } from './getNextJokeIdx';
 
-const main = async () => {
-  const joke = await jokeFetcher();
-  renderJoke(joke);
+const UPPER_BOUND = 4;
+
+const addNextJoke = async (nextJokeIdx) => {
+  const nextJoke = await jokeFetcher(nextJokeIdx);
+
+  renderJoke(nextJoke);
+};
+
+const bindEventListenerToRefreshButton = (currentJokeIdx) => {
+  const refreshButton = document.getElementById('refresh');
+  let currentJokeState = currentJokeIdx;
+
+  refreshButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+    currentJokeState = getNextJokeIdx(currentJokeState, UPPER_BOUND);
+    addNextJoke(currentJokeState);
+  });
 };
-main();
+
+const showCurrentJoke = async () => {
+  const currentJokeIdx = 0;
```

Comment:
Keeing this line in so down the line, we can replace with a random initial load index.